unnested_or_patterns trigger on master build even if feature is not enabled

When using the master toolchain and a clippy built by master, the `unnested_or_patterns` lint triggers on crates although the `or_patterns` feature is not enabled in these crates.

This does not happen if the nightly-clippy is used.

See https://github.com/rust-lang/rust-clippy/pull/5378#issuecomment-641208834 for details

Ran into this with nightly (`clippy 0.0.212 (e55d3f9c5 2020-06-18)`) now also, this clippy suggestions:

```rust
warning: unnested or-patterns
   --> components\module-host\src\host_api\world\components\component_accessors.rs:701:13
    |
701 | /             (D6Joint::RelativePosition, _)
702 | |             | (D6Joint::RelativeRotation, _)
703 | |             | (D6Joint::RelativeAngles, _) => {
    | |__________________________________________^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnested_or_patterns
help: nest the patterns
    |
701 |             (D6Joint::RelativePosition | D6Joint::RelativeRotation |
702 |  D6Joint::RelativeAngles, _) => {
```

But changing the code to this does indeed fail as it is an experimental feature we do not have enabled:

```rust
error[E0658]: or-patterns syntax is experimental
   --> components\module-host\src\host_api\world\components\component_accessors.rs:702:17
    |
702 |                 D6Joint::RelativePosition | D6Joint::RelativeRotation | D6Joint::RelativeAngles,
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: see issue #54883 <https://github.com/rust-lang/rust/issues/54883> for more information
    = help: add `#![feature(or_patterns)]` to the crate attributes to enable
```

--- a/clippy_lints/src/unnested_or_patterns.rs
+++ b/clippy_lints/src/unnested_or_patterns.rs
@@ -72,8 +72,8 @@ impl EarlyLintPass for UnnestedOrPatterns {
 }
 
 fn lint_unnested_or_patterns(cx: &EarlyContext<'_>, pat: &Pat) {
-    if !cx.sess.opts.unstable_features.is_nightly_build() {
-        // User cannot do `#![feature(or_patterns)]`, so bail.
+    if !cx.sess.features_untracked().or_patterns {
+        // Do not suggest nesting the patterns if the feature `or_patterns` is not enabled.
         return;
     }
 

--- a/tests/ui/unnested_or_patterns3.rs
+++ b/tests/ui/unnested_or_patterns3.rs
@@ -1,0 +1,6 @@
+#![warn(clippy::unnested_or_patterns)]
+
+// Test that `unnested_or_patterns` does not trigger without enabling `or_patterns`
+fn main() {
+    if let (0, 1) | (0, 2) | (0, 3) = (0, 0) {}
+}

